### PR TITLE
[DEITS] CLI: use --no prefix and fix max-size options

### PR DESCRIPTION
**Pull request by innerdvations**
### What does it do?

Replace `--encrypt <bool>` and `--compress <bool>` with `--no-encrypt` and `--no-compress`

Remove `--max-size` option for now because it hasn't been implemented yet

Allows `--max-size-jsonl` to accept a decimal value

### Why is it needed?

None of the other Strapi CLI commands use booleans, they all use default values that can be turned off with a `--no-` prefixed option.

### How to test it?

`yarn strapi export --no-compress` should now disable the compression (and `yarn strapi export --compress false` should give an error because it no longer exists as an option)

 `yarn strapi export --no-encrypt` disables the key prompt and the encryption (and `yarn strapi export --encrypt false` should give an error because it no longer exists as an option)

 `yarn strapi export --no-encrypt --key some_string` should still give an error about key only working when encryption is enabled.

 `yarn strapi export --key some_string` should work

`yarn strapi export --no-encrypt --max-size-jsonl 0.1` should split the internal files down to max 100kb (more or less)

### CODE REVIEW

File: packages/core/strapi/bin/strapi.js [hunk 1]
```diff
@@ -299,7 +289,7 @@ program
     )
   )
   .addOption(new Option('-f, --file <file>', 'name to use for exported file (without extensions)'))
-  .addOption(excludeOption)
+  // .addOption(excludeOption)
```
Comment (Convly) [inline packages/core/strapi/bin/strapi.js:292]: I think we can remove that if we're not using it. We can add it back easily if needed in the future.

File: packages/core/strapi/bin/strapi.js [hunk 2]
```diff
@@ -69,10 +68,10 @@ const getLocalScript =
 
 // option to exclude types of data for the export, import, and transfer commands
 // TODO: validate these inputs. Hopefully here, but worst case it may require adding a hook on each command
-const excludeOption = new Option(
-  '--exclude <data,to,exclude>',
-  'Comma-separated list of data to exclude (files [localMediaFiles, providerMediaFiles], content [entities, links], schema, configuration)' // ['webhooks', 'content', 'localmedia', 'providermedia', 'relations']
-).argParser(parseInputList);
+// const excludeOption = new Option(
```
Comment (Convly) [inline packages/core/strapi/bin/strapi.js:71]: Same as below, we can add it back later based on the git history, I don't think it's worth keeping this as a comment.

File: packages/core/strapi/bin/strapi.js [hunk 3]
```diff
@@ -274,18 +273,9 @@ program
...
-      `Encrypt output file using the 'aes-128-ecb' algorithm. Prompts for key unless key option is used.`
-    )
-      .default(true)
-      .argParser(parseInputBool)
-  )
-  .addOption(
-    new Option('--compress <boolean>', 'Compress output file using gzip compression')
-      .default(true)
-      .argParser(parseInputBool)
+    new Option('--no-encrypt', `Disable 'aes-128-ecb' encryption of the output file`).default(true)
```
Comment (Convly) [inline packages/core/strapi/bin/strapi.js:276]: Shouldn't it be default false now?

File: packages/core/strapi/bin/strapi.js [hunk 4]
```diff
@@ -274,18 +273,9 @@ program
...
   )
+  .addOption(new Option('--no-compress', 'Disable gzip compression of output file').default(true))
```
Comment (Convly) [inline packages/core/strapi/bin/strapi.js:278]: Same as for the --no-encrypt comment

**Review by Convly — CHANGES_REQUESTED** [review]

File: packages/core/strapi/bin/strapi.js (hunk 3 above)
Comment (christiancp100) [reply packages/core/strapi/bin/strapi.js:269]: Should we make a test for this?

**Review by Bassel17 — APPROVED** [review]
LGTM!

File: packages/core/strapi/bin/strapi.js (hunk 3 above)
Comment (innerdvations) [reply packages/core/strapi/bin/strapi.js:269]: We can't test it properly yet, that's what [this PR](https://github.com/strapi/strapi/pull/14637) is for, but we don't have time to get it in before the beta. I may change the target from `main` to `features/deits` to get it in sooner and be able to start writing tests.

Comment (Convly) [reply packages/core/strapi/bin/strapi.js:269]: > It's weird commander logic, --no- is a special prefix that treats it as if it's the value without "no-" but the default is for the value `encrypt` itself

That's so counter-intuitive lol, thanks for the information about the special prefix :slightly_smiling_face:

**Review by Convly — APPROVED** [review]

**Review by christiancp100 — APPROVED** [review]
LGTM

**Review by Convly — APPROVED** [review]
LGTM, just need to fix the conflict :slightly_smiling_face:

**Pull request merged**